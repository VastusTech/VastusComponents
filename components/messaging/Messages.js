import React from "react";
import Message from './Message';
import { Container } from "semantic-ui-react";

const Messages = (props: {messages: [any], userID: string}) => {
    return (
        <Container style={{width: '85%'}}>
            {
                props.messages.slice(0).reverse().map((message, index) => {
                    return <Message key={index} message={message} userID={props.userID}/>
                })
            }
        </Container>
    );
};

export default Messages;
