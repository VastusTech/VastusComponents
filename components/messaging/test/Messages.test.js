import React from 'react';
import { shallow } from 'enzyme';
import TestConfig, {store} from "../../../../TestConfig";
import Messages from "../Messages";

TestConfig();
it('renders without crashing', () => {
    const component = shallow(<Messages userID={null} comments={[]} store={store()}/>);
    expect(component).toMatchSnapshot();
});
