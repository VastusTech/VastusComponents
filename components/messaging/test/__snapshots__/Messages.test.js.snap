// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    comments={Array []}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    userID={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
      "style": Object {
        "width": "85%",
      },
    },
    "ref": null,
    "rendered": Array [],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
        "style": Object {
          "width": "85%",
        },
      },
      "ref": null,
      "rendered": Array [],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
